src/iarc7_motion/high_level_motion_controller.py

```diff
@@ -0,0 +1,213 @@
+                        self._action_server.set_aborted()
+            else: 
+                if self._action_server.is_canceled():
+                    self._task_command_handler.cancel_task()
+
+                self._task_command_handler.run()
+                task_state = self._task_command_handler.get_state()
+
+                # handles state of task and HLM controller
+                if isinstance(task_state, task_states.TaskCanceled):
+                    self._action_server.set_canceled()
+                    rospy.logwarn('Task was canceled')
+                    self._task = None
+                elif isinstance(task_state, task_states.TaskAborted):
+                    rospy.logwarn('Task aborted with: %s', task_state.msg)
+                    self._action_server.set_aborted()
+                    self._task = None
+                elif isinstance(task_state, task_states.TaskFailed):
+                    rospy.logwarn('Task failed with: %s', task_state.msg)
+                    self._action_server.set_succeeded(False)
+                    self._task = None
+                elif isinstance(task_state, task_states.TaskDone):
+                    self._action_server.set_succeeded(True)
+                    self._task = None
+                else:
+                    assert isinstance(task_state, task_states.TaskRunning)
+
+                # as soon as we set a task to None, start timeout timer
+                if self._task is None:
+                    self._timer = rospy.Timer(self._task_timeout, self._recieve_task_timeout)
+                    self._timeout_vel_sent = False
+
+            rate.sleep()
+
+    # fills out the Intermediary State for the task
+    def _get_transition(self):
+        state = IntermediaryState() 
+        state.last_twist = self._task_command_handler.get_last_twist()
+        state.last_task_ending_state = self._task_command_handler.get_state()
+        state.timeout_sent = self._timeout_vel_sent
+        return self._state_monitor.fill_out_transition(state)
+
+    # callback for safety task completition
+    def _safety_task_complete_callback(self, status, response):
+        with self._lock: 
+            if response.success:
+                rospy.logwarn('High Level Motion supposedly safely landed the aircraft')
+            else:
+                rospy.logerr('High Level Motion did not safely land aircraft')
+            self._safety_land_complete = True
+
+    def _recieve_task_timeout(self, event):
+        """
+        Handles no task running timeouts
+        Args: 
+            event: rospy.TimerEvent 
+                (see http://wiki.ros.org/rospy/Overview/Time)
+        """
+        with self._lock:
+            if self._task is None: 
+                if not self._timeout_vel_sent:
+                    last_twist = self._task_command_handler.get_last_twist()
+
+                    self._task_command_handler.send_timeout(
+                        self._state_monitor.get_timeout(last_twist))
```

Comment:
As I said below, I don't think this function should be in the state monitor, but also if it remains, it doesn't return a timeout. it returns an array of twiststampeds. The name should probably reflect that.

Reply:
Yes, but those twist stamps represent a timeout velocity. so maybe `get_safe_velocity` or `get_safe_timeout_vel`?

Reply:
changed to `get_timeout_twist`